### PR TITLE
Add CodeBert ONNX model

### PR DESCRIPTION
**Pull request by yiliu30**
## Type of Change

New model
API changed or not: No

## Description

Add CodeBert ONNX model

## How has this PR been tested?

Extension Test
[dynamic](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/5269/artifact/report.html)
[static](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/5258/artifact/report.html)

## Dependency Change?

The dependency changes only for this example.

### CODE REVIEW

**Comment by yiliu30** [comment]
Extension Test
[dynamic](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/5269/artifact/report.html)
[static](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/5258/artifact/report.html)

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+                "Model optimizer will be skipped. "
+                "Try to upgrade onnxruntime to avoid this error"
+            )
+            model = onnx.load(args.model_path)
+
+        config = PostTrainingQuantConfig(
+            approach="static",
+            quant_level=1,
+            quant_format=args.quant_format,
+            # recipes={"smooth_quant": True, "smooth_quant_args": {"alpha": 0.5}},
```
Comment (yuwenzho) [inline examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py:310]: need to remove `recipes`?

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        for idx, batch in enumerate(train_loader):
+            global_step += 1
+            optim.zero_grad()
+            labels = batch.pop("labels")
+            inputs = batch
+            model.train()
+            outputs = model(**inputs, labels=labels)
+            loss = outputs.loss
+            logits = outputs.logits
+            # print(np.argmax(logits.detach().numpy(), axis=1))
```
Comment (yuwenzho) [inline examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py:118]: remove the print

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+            all_labels += labels
+            all_preds += predictions
+            np.mean(
+                np.concatenate(all_labels, 0) == (np.concatenate(all_preds, 0))
+            )  # [:,0]>0.5))
+        label_flatten = np.concatenate(all_labels, 0)
+        preds_flatten = np.concatenate(all_preds, 0)
+        correct_count = np.sum(label_flatten == preds_flatten)
+        acc = correct_count / len(label_flatten)
+        print(acc)
```
Comment (yuwenzho) [inline examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py:259]: remove the print line

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+> Note: Validated ONNX Runtime [Version](/docs/source/installation_guide.md#validated-software-environment).
+
+
+## 2. Prepare Dataset
+Run `prepare_data.sh` script to download dataset from website to `dataset` folder and pre-process it:
+
+```shell
+bash prepare_data.sh
+```
+## 3. Prepare Model
```
Comment (yuwenzho) [inline examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/README.md:22]: Please update the Prepare Model step to `python prepare_model.py --input_model=a --output_model=b` format

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+torch
+transformers
+accelerate
+onnx
+onnxruntime
+coloredlogs
+sympy
+onnxruntime-extensions; python_version < '3.10'
```
Comment (yuwenzho) [inline examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/requirements.txt:8]: onnxruntime-extensions is available for python 3.10 now. Please update < '3.10' to < '3.11'

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+        dataloader = INC_DataLoader(
+            framework="onnxruntime", dataset=ort_dataset, batch_size=args.batch_size
+        )
+        model = onnx.load(args.model_path)
+        if args.mode == "performance":
+            from neural_compressor.benchmark import fit
+            from neural_compressor.config import BenchmarkConfig
+
+            conf = BenchmarkConfig(
+                iteration=100, cores_per_instance=28, num_of_instance=1
```
Comment (mengniwang95) [inline examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py:337]: add parameters to make cores_per_instance and num_of_instance are configurable

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/run_benchmark.sh [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+# run_benchmark
+function run_benchmark {
+
+    python main.py \
+            --model_name_or_path microsoft/codebert-base \
+            --model_path ${input_model} \
+            --data_path ${dataset_location} \
+            --mode=${mode} \
+            --batch_size=${batch_size} \
```
Comment (mengniwang95) [inline examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/run_benchmark.sh:41]: ermove "=", align with above parameter

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/run_quant.sh [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      --output_model=*)
+          output_model=$(echo $var |cut -f2 -d=)
+      ;;
+      --input_model=*)
+          input_model=$(echo $var |cut -f2 -d=)
+      ;;
+      --dataset_location=*)
+          dataset_location=$(echo $var |cut -f2 -d=)
+      ;;
+      --batch_size=*)
```
[the remark this answers is not in the document]
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/run_quant.sh]: Thx, corrected by [6df25bb](https://github.com/intel/neural-compressor/pull/1121/commits/6df25bb0d6cd251a471bb8a445033390458021ca), btw the dynamic should not include `--quant_format`?

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py (hunk 1 above)
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py]: Removed by [0c26397](https://github.com/intel/neural-compressor/pull/1121/commits/0c2639750f33f6b306544dba11175291c4ad691d).

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py (hunk 2 above)
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py]: Removed by [0c26397](https://github.com/intel/neural-compressor/pull/1121/commits/0c2639750f33f6b306544dba11175291c4ad691d).

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py (hunk 3 above)
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py]: Removed by [0c26397](https://github.com/intel/neural-compressor/pull/1121/commits/0c2639750f33f6b306544dba11175291c4ad691d).

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py (hunk 6 above)
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py]: Thx, updated by [0c26397](https://github.com/intel/neural-compressor/pull/1121/commits/0c2639750f33f6b306544dba11175291c4ad691d).

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/run_benchmark.sh (hunk 7 above)
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/run_benchmark.sh]: Updated by [6df25bb](https://github.com/intel/neural-compressor/pull/1121/commits/6df25bb0d6cd251a471bb8a445033390458021ca).

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/requirements.txt (hunk 5 above)
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/requirements.txt]: Updated by [345c1b5](https://github.com/intel/neural-compressor/pull/1121/commits/345c1b5c62ae066fbc88596ea7737988f927f8dd).

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/README.md (hunk 4 above)
Comment (yiliu30) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/README.md:22]: Will update it with another PR after the model upload and `prepare_model.py` template are finalized :)

**Comment by yiliu30** [comment]
Rerun the EXT test [5272](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/5272/).

File: examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/run_quant.sh (hunk 8 above)
[the remark this answers is not in the document]
Comment (yuwenzho) [reply examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_dynamic/run_quant.sh]: yes, dynamic should not include `--quant_format`

**Review by yuwenzho — APPROVED** [review]

**Review by chensuyue — APPROVED** [review]

**Pull request merged**